network-config-analyzer/IstioPolicyYamlParser.py

```diff
@@ -64,39 +65,106 @@ def parse_label_selector(self, label_selector):
+        elif field_name == 'principals':
+            values_list = self._get_all_principals_str_list_from_topology()
+        elif field_name == 'methods':
+            values_list = MethodSet.all_methods_list
+        if '*' not in regex_str:  # exact match
+            return [regex_str] if regex_str in values_list else []
+        elif regex_str == '*':  # presence match
+            return values_list
+        elif regex_str.startswith('*'):  # Suffix match
+            required_suffix = regex_str.replace('*', '')
+            return [val for val in values_list if val.endswith(required_suffix)]
+        elif regex_str.endswith('*'):  # Prefix match
+            required_prefix = regex_str.replace('*', '')
+            return [val for val in values_list if val.startswith(required_prefix)]
+        return []
+
+    def _parse_ns_str(self, ns):
         """
         parse a namespace string from source component in rule
         :param str ns: the namespace string
         :return: PeerSet: All pods in the namespace ns
         """
-        # TODO: value matching should be extended to regular expressions patterns
-        # TODO: when supporting this, can add validation that * is only used as prefix/suffix
-        if '*' in ns:
-            self.syntax_error(
-                'error parsing namespace regular expr: currently not supporting prefix/suffix patterns ', ns)
-        ns_obj = self.peer_container.get_namespace(ns)
-        return self.peer_container.get_namespace_pods(ns_obj)
+        ns_str_values = self._parse_istio_regex_from_enumerated_domain(ns, 'namespaces')
+        res = PeerSet()
+        if not ns_str_values:
+            self.warning(f"no match for namespace: {ns}")
+        for ns_str in ns_str_values:
+            ns_obj = self.peer_container.get_namespace(ns_str)
+            res |= self.peer_container.get_namespace_pods(ns_obj)
+        return res
 
-    def parse_principal_str(self, principal):
+    @staticmethod
+    def _get_principal_str_components(principal_str):
         """
-        parse a principal string from source component in rule
-        :param str principal: the principal str, currently assuming in format: "cluster.local/ns/<ns-str>/sa/<sa-str>"
-        :return: PeerSet: All pods with the given ns + sa_name as extracted from principal str
+        get the namespace and service account values from a principal str
+        :param str principal_str: the principal str
+        :return: the namespace and service account str values
         """
-        if '*' in principal:
-            self.syntax_error(
-                'error parsing principal regular expr: currently not supporting prefix/suffix patterns ', principal)
+        ns = None
+        sa_name = None
         # principal_str_example = "cluster.local/ns/default/sa/sleep"
         # TODO: support a more general pattern for principal str (prefix by istio trust-domain)
         # TODO: tighter checks in parsing the principal str
```

Comment:
Are these two TODOs still relevant?

Reply:
The second one is not relevant, the first one is still relevant. Updated the comments here.